### PR TITLE
WidgetLearnerTest: GuiToParam refactor

### PR DESCRIPTION
**Pull request by VesnaT**

### CODE REVIEW

File: Orange/widgets/tests/base.py [hunk 1]
```diff
@@ -175,7 +176,77 @@ def get_output(self, output_name, widget=None):
         return self.signal_manager.outputs.get((widget, output_name), None)
 
 
-GuiToParam = namedtuple('GuiParam', 'name gui_el get set values set_values')
+class GuiToParam:
```
Comment (kernc) [inline Orange/widgets/tests/base.py]: Missing docstring.

File: Orange/widgets/tests/base.py [hunk 2]
```diff
@@ -176,7 +176,119 @@ def get_output(self, output_name, widget=None):
...
+    """Class for mapping between gui components and learner's parameters
+    when testing unchecked properties and therefore default parameters
+    should be used.
+
+    Parameters
+    ----------
+    name : str
+        Name of learner's parameter.
+
+    gui_element : QWidget
```
Comment (astaric) [inline Orange/widgets/tests/base.py]: Why is gui_element needed as parameter if its value will never be used?

File: Orange/widgets/tests/base.py [hunk 3]
```diff
@@ -176,7 +176,119 @@ def get_output(self, output_name, widget=None):
...
+        Gui component who's corresponding parameter is to be tested.
+
+    default_value: str, int,
+        Value that learner uses by default.
```
Comment (astaric) [inline Orange/widgets/tests/base.py]: "Value that should be used by default"

Widgets might decide to provide different defaults from the underlying learners.

File: Orange/widgets/tests/base.py [hunk 4]
```diff
@@ -176,7 +176,119 @@ def get_output(self, output_name, widget=None):
...
+                         values or self._default_values(gui_element),
+                         getter or self._default_get_value(gui_element, values),
+                         setter or self._default_set_value(gui_element, values))
+
+    @staticmethod
+    def _default_values(gui_el):
+        if isinstance(gui_el, (QSpinBox, QDoubleSpinBox, QSlider)):
+            return [gui_el.minimum(), gui_el.maximum()]
+        else:
+            raise
```
Comment (astaric) [inline Orange/widgets/tests/base.py]: This does not work. You need to specify the exception to be raised.

File: Orange/widgets/tests/base.py [hunk 5]
```diff
@@ -300,24 +411,25 @@ def test_parameters_default(self):
         self.widget.apply_button.button.click()
         if hasattr(self.widget.learner, "params"):
             learner_params = self.widget.learner.params
-            for element in self.gui_to_params:
+            for element in self.parameters:
```
Comment (astaric) [inline Orange/widgets/tests/base.py]: I would rename this to

    for parameter in self.parameters:

to match similar for in test_parameters method

File: Orange/widgets/classify/tests/test_owlogisticregression.py [hunk 6]
```diff
@@ -37,21 +37,22 @@ def setUp(self):
         self.widget = self.create_widget(OWLogisticRegression,
                                          stored_settings={"auto_apply": False})
         self.init()
+        c_slider = self.widget.c_slider
 
-        def combo_set_value(i, x):
-            x.activated.emit(i)
-            x.setCurrentIndex(i)
+        def get_fun():
```
Comment (astaric) [inline Orange/widgets/classify/tests/test_owlogisticregression.py]: could this be renamed to getter?

File: Orange/widgets/classify/tests/test_owlogisticregression.py [hunk 7]
```diff
@@ -37,21 +37,22 @@ def setUp(self):
...
-        c_slider = self.widget.c_slider
-        c_min_max = [c_slider.minimum(), c_slider.maximum()]
-        c_min_max_values = [self.widget.C_s[0], self.widget.C_s[-1]]
-        self.gui_to_params = [
-            GuiToParam('penalty', self.widget.penalty_combo,
-                       lambda x: pen_types[x.currentIndex()],
-                       combo_set_value, pen_types, list(range(len(pen_types)))),
-            GuiToParam('C', c_slider, lambda x: self.widget.C_s[x.value()],
-                       lambda i, x: x.setValue(i), c_min_max_values, c_min_max)]
+        def set_fun(val):
```
Comment (astaric) [inline Orange/widgets/classify/tests/test_owlogisticregression.py]: and this to setter?

File: Orange/widgets/tests/base.py (hunk 2 above)
Comment (astaric) [inline Orange/widgets/tests/base.py]: gui_element was removed from DefaultParameterMapping so this is considered fixed.

**Comment by astaric** [comment]
I have no further comments. If you can rebase it onto the latest master, I will merge.

**Pull request merged**